src/protobuf/v1/worker/worker.proto

```diff
@@ -18,6 +18,7 @@ message WorkerRequest {
     string namespace = 4;
     string clustername = 5;
     string labels = 6;
+    string fromsource = 7;
```

Comment:
This change should not be part of this PR. Please rebase the code.

Reply:
Okay